common/basecomponent/base.go

```diff
@@ -220,8 +322,13 @@ func (b *BaseDendrite) SetupAndServeHTTP(bindaddr string, listenaddr string) {
 	logrus.Infof("Stopped %s server on %s", b.componentName, addr)
 }
 
-// setupKafka creates kafka consumer/producer pair from the config.
+// setupKafka creates kafka consumer/producer pair from the config. Checks if
+// should use naffka.
 func setupKafka(cfg *config.Dendrite) (sarama.Consumer, sarama.SyncProducer) {
+	if cfg.Kafka.UseNaffka {
+		return setupNaffka(cfg)
```

Comment:
Why was this moved? There's a lack of symmetry now because:
 - `setupKafka` will check the config and use Naffka if it has been told to do so.
 - `setupNaffka` will NOT check the config and use Kafka if it has been told to do do.

What was wrong with it where it previously was?

Reply:
I've set base.go to be identical with the one upstream, and it works fine without these changes. :shrug: